packages/blockchain-api/src/blockscout.ts

```diff
@@ -44,33 +45,31 @@ export class BlockscoutAPI extends RESTDataSource {
     const { celoTransfers } = transferTx
 
     const transfersCount = celoTransfers.length
-    if (transfersCount < 3) {
+    // 3 fee transfers when gatewayFeeRecipient is set, otherwise 2
+    const expectedFeeTransfersCount = transferTx.gatewayFeeRecipient ? 3 : 2
+
+    if (transfersCount < expectedFeeTransfersCount) {
       throw new Error(`Cannot determine fee transfers for tx ${transferTx.transactionHash}`)
     }
 
-    const feeTransfer1 = celoTransfers[transfersCount - 1]
-    const feeTransfer2 = celoTransfers[transfersCount - 2]
-    const feeTransfer3 = celoTransfers[transfersCount - 3]
-    const fee1Value = new BigNumber(feeTransfer1.value)
-    const fee2Value = new BigNumber(feeTransfer2.value)
-    const fee3Value = new BigNumber(feeTransfer3.value)
-    const gasValue = new BigNumber(transferTx.gasUsed).multipliedBy(transferTx.gasPrice)
-    const gatewayFeeValue = new BigNumber(transferTx.gatewayFee)
-    const expectedTotalFeeValue = gasValue.plus(gatewayFeeValue)
+    let totalFee = new BigNumber(0)
+    for (let i = 1; i <= expectedFeeTransfersCount; i++) {
```

Comment:
Should we use `i <= transfersCount;` here?
Otherwise, in case `transfersCount > expectedFeeTransfersCount` we would not count all the fees, right?

Reply:
I think the logic is correct.
The fee transfers are always the last 2 or 3 transfers.

Let's say it's the case with 3 fee transfers, we have the following transfers for the transaction we're looking at:
```
[
  transfer1,
  transfer2,
  feeTransfer1,
  feeTransfer2
  feeTransfer3
]
```
We want to add only `feeTransfer1`, `feeTransfer2`, `feeTransfer3`, so we iterate from 1 to 3 taking transfers from the end of the array.

Does this make it clearer for you?

Reply:
Yeah, thanks!